packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -907,11 +945,19 @@ class _SearchAnchorWithSearchBar extends SearchAnchor {
     super.isFullScreen,
     super.searchController,
     super.textCapitalization,
+    ValueChanged<String>? viewOnChanged,
+    ValueChanged<String>? viewOnSubmitted,
```

Comment:
`viewOnChanged` and `viewOnSubmitted` give the user control over the "search view" experience. 
But, the user doesn't have control over the "search bar" experience. My thoughts are that either

-  `barOnChanged` and `barOnSubmitted` properties are added
or 
- `viewOnChanged` and `viewOnSubmitted` are replaced with more generic `onChanged` and `onSubmitted` which are applied to both the "search bar" and "search view" experiences.

Reply:
Thanks a lot for sharing these thoughts! I think `barOnChanged` and `barOnSubmitted` might be not necessary because when we tap the bar, the view will open and the search bar itself will lose focus, then what we really control next is the search bar on the search view.

Renaming `viewOnChanged` and `viewOnSubmitted` sounds good but I thought it might also be good that we can keep consistent between the parameter names of the factory method and the property names of `SearchAnchor`. Does this make sense?

Reply:
Thanks @QuncCccccc for taking the time to get this right. I understand what you are saying and you make good points. Please allow me to present a different perspective though. 

For SearchAnchor.bar, while it's true that tapping the bar causes the "search view" to open; it's possible to then dismiss the "search view", return to the "search bar", and then submit. So this is a case where the "search bar" needs to listen for a submit.  

I think that using the `view*` prefix makes sense if `viewOnSubmitted` only applies to the "search view" experience. But as I just described, `SearchAnchor.bar`'s `viewOnSubmitted` would need to apply to both "search view" and "search bar" experiences; so I think it can be named more generically like `onSubmitted`.   

And while you could name the property `viewOnChanged` in `SearchAnchor.bar`, it feels a bit like exposing an implementation detail. I mean, all the user of `SearchAnchor.bar` cares about is that they get callbacks when the text changes. The user of `SearchAnchor.bar` doesn't care that the implementation is in the "search view", right? 
Also, the reason why `viewOnChanged` in `SearchAnchor` makes sense is because the `view*` prefix makes clear that it won't affect the "search bar" experience. In the case of `SearchAnchor.bar` factory, we can go with the simpler property name (`onChanged`) because there isn't the same opportunity for confusion.

So this leads me to this signature: 

SearchAnchor.bar( onSubmitted: ..., onChanged: ..., ...)

What do you think?

Reply:
> For `SearchAnchor.bar`, while it's true that tapping the bar causes the "search view" to open; it's possible to then dismiss the "search view", return to the "search bar", and then submit. So this is a case where the "search bar" needs to listen for a submit.

Oh I think this is a reasonable use case. Adding `barOnSubmitted` for `SearchAnchor.bar`

Reply:
Wait, you've misunderstood my point. I'm saying that because the onSubmitted will apply to both the "search bar" and the "search view", that we don't need separate parameters for each one, and can combine them into a single parameter.

Why would someone need separate barOnSubmitted and viewOnSubmitted parameters in this case?

Reply:
> Why would someone need separate barOnSubmitted and viewOnSubmitted parameters in this case?

I was thinking if a use case is possible that developers just want different `onSubmit` behaviors for the two search bars. If that's possible then probably we should just provide two parameters. But that might sound a little unusual...

Reply:
> I was thinking if a use case is possible that developers just want different onSubmit behaviors for the two search bars.

If developers want different onSubmit behaviors then they can use the plain SearchAnchor constructor, right? 

To me, the SearchAnchor.Bar factory appears designed to be an easier to use constructor with reasonable defaults setup automatically.

Reply:
This makes sense to me. Thanks for the suggestion! Updated. 

I also added `onChanged` for `SearchAnchor.bar`.